k8s.go

```diff
@@ -0,0 +1,245 @@
+package profitbricks
+
+import "net/http"
+
+type KubernetesClusters struct {
+	// URL to the collection representation (absolute path)
+	// Read Only: true
+	// Format: uri
+	Href string `json:"href,omitempty"`
+
+	// Unique representation for Kubernetes Cluster as a collection on a resource.
+	// Read Only: true
+	ID string `json:"id,omitempty"`
+
+	// Slice of items in that collection
+	// Read Only: true
+	Items []KubernetesCluster `json:"items"`
+
+	// The type of resource within a collection
+	// Read Only: true
+	// Enum: [collection]
+	PBType string `json:"type,omitempty"`
+}
+
+// KubernetesCluster kubernetes cluster
+type KubernetesCluster struct {
+	// URL to the object representation (absolute path)
+	// Read Only: true
+	// Format: uri
+	Href string `json:"href,omitempty"`
+
+	// The resource's unique identifier.
+	// Read Only: true
+	ID string `json:"id,omitempty"`
+
+	// metadata
+	Metadata *Metadata `json:"metadata,omitempty"`
+
+	// properties
+	// Required: true
+	Properties *KubernetesClusterProperties `json:"properties"`
+
+	// The type of object
+	// Read Only: true
+	// Enum: [k8s]
+	PBType string `json:"type,omitempty"`
+
+	// Entities of a cluster
+	Entities KubernetesClusterEntities `json:"entities"`
+}
+
+type KubernetesClusterEntities struct {
+	// NodePools of a cluster
+	NodePools *KubernetesNodePools `json:"nodepools,omitempty"`
+}
+
+// KubernetesClusterProperties kubernetes cluster properties
+type KubernetesClusterProperties struct {
+	// A Kubernetes Cluster Name. Valid Kubernetes Cluster name must be 63 characters or less and must not be empty
+	// and begin and end with an alphanumeric character ([a-z0-9A-Z]) with dashes (-), dots (.),
```

Comment:
I looked up the underscore issue again. Underscores are fine, the docs are correct.

You confused the restrictions applying to label/annotation **keys' prefix parts**.  The keys' name parts can include underscores. The values, too.
The restriction applying to the values are identical to the restrictions for cluster and nodepool names, so underscores are fine.
(Please also think of informing Ionos again that your previous information about underscores being invalid was not correct.)

Reply:
There's something I need to add: The cluster and nodepool names were intended to be DNS labels but the regex in place allows too much, making the names actually K8s qualified names. So underscores are allowed here, even though DNS wouldn't allow them.